python/cudf/cudf/tests/test_orc.py

```diff
@@ -670,3 +673,29 @@ def test_orc_reader_gmt_timestamps(datadir):
     pdf = orcfile.read().to_pandas()
     gdf = cudf.read_orc(path, engine="cudf").to_pandas()
     assert_eq(pdf, gdf)
+
+
+def test_orc_bool_encode_fail():
+    np.random.seed(0)
+
+    # Generate a boolean column longer than a single stripe
+    fail_df = cudf.DataFrame({"col": gen_rand_series("bool", 600000)})
+    # Invalidate the first row in the second stripe to break encoding
+    fail_df["col"][500000] = None
+
+    # Should throw instead of generating a file that is incompatible
+    # with other readers (see issue #6763)
+    with pytest.raises(RuntimeError):
+        fail_df.to_orc("should_throw.orc")
+
+    # Generate a boolean column that fits into a single stripe
+    okay_df = cudf.DataFrame({"col": gen_rand_series("bool", 500000)})
+    okay_df["col"][500000 - 1] = None
```

Comment:
Can we just use `okay_df["col"][-1]` here?

Reply:
I expected the same, but it didn't work when I tried.

Reply:
@isVoid can you check to see if that's a bug related to scatter?

Reply:
:man_facepalming: I can't write it like that because `-1` is not a label in the index.. In pandas this assigns to a new row:
```python
>>> s = pd.Series([1, 2, 3])
>>> s[-1] = None
>>> s
 0       1
 1       2
 2       3
-1    None
dtype: object
```
Worth noting that this use case is not supported in cudf. But something like `s.iloc[-1] = None` would serve the purpose here.

Reply:
Could you raise an issue about this, even if just for tracking?

Reply:
Raised: #7290